src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/PrivacyNoticesServiceImpl.scala

```diff
@@ -0,0 +1,56 @@
+package it.pagopa.interop.backendforfrontend.service.impl
+
+import cats.syntax.all._
+import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
+import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.interop.commons.utils.TypeConversions._
+import org.scanamo.DynamoReadError.describe
+import org.scanamo._
+import org.scanamo.syntax._
+import it.pagopa.interop.backendforfrontend.service.PrivacyNoticesService
+import it.pagopa.interop.backendforfrontend.error.BFFErrors.DynamoReadingError
+import it.pagopa.interop.backendforfrontend.service.model.{PrivacyNotice, UserPrivacyNotice}
+
+import java.util.UUID
+import scala.concurrent.{Future, ExecutionContext}
+
+class PrivacyNoticesServiceImpl(tableName: String)(implicit ec: ExecutionContext, scanamo: ScanamoAsync)
+    extends PrivacyNoticesService {
+
+  implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  val pnTable: Table[PrivacyNotice] =
+    Table[PrivacyNotice](tableName)
+
+  val userTable: Table[UserPrivacyNotice] =
+    Table[UserPrivacyNotice](tableName)
+
+  override def getLatestVersion(id: UUID)(implicit contexts: Seq[(String, String)]): Future[Option[PrivacyNotice]] = {
+    logger.info(s"Getting id $id privacy notice")
+    scanamo
+      .exec { pnTable.get("pk" === s"${PrivacyNotice.pkPrefix}$id" and "sk" === s"${PrivacyNotice.skPrefix}$id") }
```

Comment:
have the prefix a purpose that I'm missing right now?
Could a proper field naming remove the need of the prefix? Thinking something like

```suggestion
      .exec { pnTable.get("privacyNoticeId" === id) }
```

(btw the sort key for this table should be the version, but the query is done with the privacy notice id)

Reply:
I used a single table pattern to simulate a one to many cardinality on NoSQL DB. In this case the latest version is the only version available for PN id. It is weird to see it in the code but, trust me, it is the right pattern.

Reply:
ok, can you elaborate a little more more on the reason why this is the preferable solution?

Reply:
Because we need to persist the latest version and I cannot insert into the primary key (in this case in the sort key) the version, otherwise I have multiple records for multiple version for the same privacy notices.